themes/src/main/resources/theme/base/login/user-profile-commons.ftl

```diff
@@ -54,6 +58,29 @@
 		<#nested "afterField" attribute>
 	</#list>
 
+	<script type="module">
+		const attributesMap = new Map();
+		<#list profile.attributes as attribute>
+		const ${attribute.name?replace("-", "_")}Attr = {};
+		${attribute.name?replace("-", "_")}Attr.getName = () => "${attribute.name}";
+		${attribute.name?replace("-", "_")}Attr.getDisplayName = () => {
+			const displayName = "${attribute.displayName}";
+
+			if (displayName) {
+				return displayName;
+			}
+
+			return "${attribute.name}";
+		}
+		attributesMap.set("${attribute.name}", ${attribute.name?replace("-", "_")}Attr);
+		</#list>
+		const profile = {};
+		profile.getAttributes = () => { return attributesMap; };
+		profile.getAttribute = (name) => { return attributesMap.get(name); };
+		window.keycloak = {};
+		window.keycloak.profile = profile;
```

Comment:
These globals should not be defined here, but instead use a module to do so.

Reply:
How to dynamically build a module based on the user profile metadata without introducing changes to our backend?

Reply:
I've removed this from the PR so we can enable exposing metadata to user-defined modules as a follow-up.